src/Configuration/ConfigurationResolver.php

```diff
@@ -30,7 +30,8 @@ public static function resolve(
         $key,
         $defaultValue,
         $expectedType,
-        $config = []
+        $config = [],
```

Comment:
Aren't config and options _basically_ synonyms?  Why are we differentiating them here?  Seems like we could flatten this into one array

Reply:
`config` is passed in from the client args/config in the client constructor, but I suppose we could flatten additional options into it.